pallets/connectors/src/lib.rs

```diff
@@ -427,13 +453,89 @@ pub mod pallet {
 			)?;
 
 			Self::do_send_message(
-				who,
-				Message::Transfer {
+				who.clone(),
+				Message::TransferTrancheTokens {
 					pool_id,
 					tranche_id,
 					amount,
 					domain: domain_address.domain(),
-					address: domain_address.address(),
+					receiver: domain_address.address(),
+					sender: who
+						.encode()
+						.try_into()
+						.map_err(|_| DispatchError::Other("Conversion to 32 bytes failed"))?,
+				},
+				domain_address.domain(),
+			)?;
+
+			Ok(())
+		}
+
+		/// Transfer non-tranche tokens to a given address
+		#[pallet::weight(< T as Config >::WeightInfo::transfer())]
+		#[pallet::call_index(7)]
+		pub fn transfer(
+			origin: OriginFor<T>,
+			asset_id: CurrencyIdOf<T>,
+			domain_address: DomainAddress,
+			amount: <T as pallet::Config>::Balance,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			ensure!(!amount.is_zero(), Error::<T>::InvalidTransferAmount);
+
+			// Transfer to the domain account for bookkeeping
+			T::Tokens::transfer(
+				asset_id,
+				&who,
+				&DomainLocator::<Domain> {
+					domain: domain_address.domain(),
+				}
+				.into_account_truncating(),
+				amount,
+				false,
+			)?;
+
+			Self::do_send_message(
+				who.clone(),
+				Message::Transfer {
+					amount,
+					currency: <Pallet<T> as GeneralCurrencyIndex<
+						<T as Config>::GeneralCurrencyPrefix,
+					>>::get_general_index(asset_id)?,
+					sender: who
+						.encode()
+						.try_into()
+						.map_err(|_| DispatchError::Other("Conversion to 32 bytes failed"))?,
+					receiver: domain_address.address(),
+				},
+				domain_address.domain(),
+			)?;
+
+			Ok(())
+		}
+
+		/// Add a CurrencyId to the set of known currencies, enabling that currency for Pool creation.
```

Comment:
```suggestion
		/// Add a CurrencyId to the set of known currencies on a given Domain
```

The "enabling that currency for Pool creation" is not necessarily correct for two reasons:
1. One can't create a pool through a Connector Domain, so adding a currency to a Domain doesn't enable that currency for pool creation
2. It's through `AllowPoolCurrency` that we enable a currency to be used as a pool currency and for investments etc